app.js

```diff
@@ -1,7 +1,29 @@
+import { signInUser, signupUser, redirectIfLoggedIn } from './fetch-utils.js';
+
 const signInForm = document.getElementById('sign-in');
 
 const signUpForm = document.getElementById('sign-up');
 
 // Wire up sign in and sign up forms to supabase
 // Redirect to /other-page on successful auth
 // Redirect to /other-page when page loads if user is authenticated
+
+signUpForm.addEventListener('submit', async (e) => {
+    e.preventDefault();
+
+    const data = new FormData(signUpForm);
+
+    await signupUser(data.get('email'), data.get('password'));
+
+    redirectIfLoggedIn();
+});
+
+signInForm.addEventListener('submit', async (e) => {
+    e.preventDefault();
+
+    const data = new FormData(signInForm);
+
+    await signInUser(data.get('email'), data.get('password'));
+
+    redirectIfLoggedIn();
+});
```

Comment:
somewhere in this file you should call redirectIfLoggedIn() so that users who are logged in, cannot go to the auth page to log in :D